internal/traceprof/traceproftest/traceprof_test.go

```diff
@@ -51,20 +51,20 @@ func TestEndpointsAndCodeHotspots(t *testing.T) {
 
 			notEnoughSamples := (prof.Duration() < minCPUDuration) ||
 				(prof.LabelsDuration(CustomLabels) < minCPUDuration) ||
-				(c.CodeHotspots && prof.LabelDuration(traceprof.SpanID, "*") < minCPUDuration) ||
-				(c.AppType != Direct && c.Endpoints && prof.LabelDuration(traceprof.TraceEndpoint, "*") < minCPUDuration)
-			if attempt <= 5 && notEnoughSamples {
+				(c.CodeHotspots && prof.LabelsDuration(map[string]string{traceprof.LocalRootSpanID: res.LocalRootSpanId, traceprof.SpanID: res.SpanId}) < minCPUDuration) ||
+				(c.Endpoints && prof.LabelDuration(traceprof.TraceEndpoint, c.AppType.Endpoint()) < minCPUDuration)
```

Comment:
I think removing `c.AppType != Direct` is the only change here that is needed for fixing the flakiness of the test. It should have been done as part of this commit, but I forgot: https://github.com/DataDog/dd-trace-go/commit/65599f2e8a1dd1bfba3be4f1667c008ef26a28ba

The other changes seem reasonable as well, but not necessary. I'm okay with merging this as-is or merging a smaller PR that just removes the `c.AppType != Direct` expression. Up to you : ).

Reply:
I've seen the test fail for non-`Direct` app types ([example](https://app.circleci.com/pipelines/github/DataDog/dd-trace-go/2642/workflows/a8524936-d9c0-49cf-a88f-8b4eb7d71d4f/jobs/18826), and I was also able to make the grpc app type fail locally). I think the underlying issue is that the test did not check for all of the expected labels when deciding whether to re-try, so it could exit early with some of the labels while still missing others.

Reply:
I hadn't seen those other failures. Ok, in that case I think you're right. Let's go ahead with all your suggested changes in here!